### DISCUSSION TITLE
How to install LazyVim with plugins offline?

### DISCUSSION THREAD

**Discussion by ormeilu** [LazyVim/LazyVim#3745] [Q&A]
# It's time to remove all my IDEs?

## My problem

Hi! I'm thinking about dropping my IDEs because they are all bad with:
1. Installing and updating IDEs offline
2. Adding and upgrading plugins offline
3. Remote developing without internet access

As you can see - all my problems comes from working in closed from internet environment.

To make VSCode working - I just need to install .vsix plugin files. But when it comes to developing trough ssh connections - VSCode tries to install VSCode Server via internet and (oh wow) fails. So I need to manually install server every time and that process disguisting.

With JetBrains (PyCharm/DataSpell) its the same process, but server-side is not fixable - I just wasn't able to fix server installation. Only remote interpreter works, but not ssh remote development.

## What do I want?

I want to just connect to remote ssh inside closed from internet environment and start developing. I don't want to mess with installation and other stupid things like downloading and manually installing IDE's server for remote development.

## My guess how LazyVim can fix that
My guess is that installing nvim, some other dependecies (that not a problem for me) and copying `.nvim` config files would be enough to have fully working nvim setup. Am I Right?

If not - how to do this?

**Comment by DrummyFloyd** [comment]
i see 2 solutions depends on your need 

* sshfs => drop the working dir trhoug ssh into your local machine

* https://github.com/amitds1997/remote-nvim.nvim => never tried ( but support offline mode)

**Reply by ormeilu** [reply]
1. looks loke `sshfs` might be a good idea (even for windows I've found a port), I'll try that
2. I have both systems (local and remote) in offline mode and some of the plugins like nvim-treesitter requires internet (but it can be fixed with some tinkering https://github.com/nvim-treesitter/nvim-treesitter/discussions/4423#discussioncomment-5179505 )

I'll try both methods and then comeback to you with a result. Thank you!
